frontend/taipy-gui/src/context/taipyReducers.ts

```diff
@@ -812,16 +819,22 @@ const getAlertType = (aType: string) => {
     return aType;
 };
 
+const generateUniqueId = () => {
+    return '_' + Math.random().toString(36).substr(2, 9);
+};
+
 export const createAlertAction = (alert: AlertMessage): TaipyAlertAction => ({
     type: Types.SetAlert,
     atype: getAlertType(alert.atype),
     message: alert.message,
     system: alert.system,
     duration: alert.duration,
+    notification_id: alert.notification_id || generateUniqueId(),
```

Comment:
What do we need a random identifier for?

Reply:
to automatically delete the notification from the state.
before adding notification_id, we got the id generated by notistack